lua/autorun/client/cl_mcsp.lua

```diff
@@ -0,0 +1,36 @@
+local MCSPmodel = "models/milaco/minecraft_pm/minecraft_pm.mdl"
+
+hook.Add("PrePlayerDraw", "MCSPRenderSkin", function(ply)
+	if IsValid(ply) and ply:GetModel() == MCSPmodel and ply:IsPlayer() then
+		render.MaterialOverrideByIndex(0, ImgurMaterial(ply:GetNWString("MCSPSkinURL", false), ply, ply:GetPos(), false, "VertexLitGeneric", {
+			["$alphatest"] = 1,
+			["$halflambert"] = 1,
+			["$nodecal"] = 1,
+			["$model"] = 1
+		}))
+	end
+end)
+
+hook.Add("PostPlayerDraw", "MCSPRenderSkin", function(ply)
+	if IsValid(ply) and ply:GetModel() == MCSPmodel and ply:IsPlayer() then
+		render.MaterialOverrideByIndex(0, "")
+	end
+end)
+
+hook.Add("PostDrawOpaqueRenderables", "DrawMinecraftPlayerRagdolls", function() --Renders the material on the player's ragdoll
+	for _, v in pairs(player.GetHumans()) do
+		if IsValid(v) and IsValid(v:GetRagdollEntity()) and v:GetRagdollEntity():GetModel() == MCSPmodel then
+			local ragent = v:GetRagdollEntity()
+
+			render.MaterialOverrideByIndex(0, ImgurMaterial(ply:GetNWString("MCSPSkinURL", false), ply, ply:GetPos(), false, "VertexLitGeneric", {
+				["$alphatest"] = 1,
+				["$halflambert"] = 1,
+				["$nodecal"] = 1,
+				["$model"] = 1
+			}))
+			ragent:DrawModel()
+			ragent:SetNoDraw(true)
```

Comment:
can you please test that the ragdolls still work if you change your model after you respawn? ie it doesnt use the same entity which is still NoDraw'd next time you die

Reply:
I tried my best to recreate ImgurMaterial and tested the skins on a bunch of bots constantly changing skin URLs. They kept the URLs even after death and ragdolling a second time, so I'm assuming it works. I copied this bit of code over from one of the minecraft skin pickers on the workshop.

Reply:
okay thanks, eventually i will get another repository up for common code like ImgurMaterial